pkg/controller/imagelistpulljob/imagelistpulljob_controller.go

```diff
@@ -0,0 +1,423 @@
+func newReconciler(mgr manager.Manager) *ReconcileImageListPullJob {
+	return &ReconcileImageListPullJob{
+		Client:   utilclient.NewClientFromManager(mgr, controllerName),
+		scheme:   mgr.GetScheme(),
+		clock:    clock.RealClock{},
+		recorder: mgr.GetEventRecorderFor(controllerName),
+	}
+}
+
+// add a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r *ReconcileImageListPullJob) error {
+	// Create a new controller
+	c, err := controller.New(controllerName, mgr, controller.Options{Reconciler: r, MaxConcurrentReconciles: concurrentReconciles})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to ImageListPullJob
+	err = c.Watch(&source.Kind{Type: &appsv1alpha1.ImageListPullJob{}}, &handler.EnqueueRequestForObject{}, predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			oldObj := e.ObjectOld.(*appsv1alpha1.ImageListPullJob)
+			newObj := e.ObjectNew.(*appsv1alpha1.ImageListPullJob)
+			return oldObj.Generation != newObj.Generation
+		},
+	})
+	if err != nil {
+		return err
+	}
+	// Watch for changes to ImagePullJob
+	err = c.Watch(&source.Kind{Type: &appsv1alpha1.ImagePullJob{}}, &imagePullJobEventHandler{
+		enqueueHandler: handler.EnqueueRequestForOwner{
+			IsController: true,
+			OwnerType:    &appsv1alpha1.ImageListPullJob{},
+		},
+	})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+var _ reconcile.Reconciler = &ReconcileImageListPullJob{}
+
+// ReconcileImageListPullJob reconciles a ImageListPullJob object
+type ReconcileImageListPullJob struct {
+	client.Client
+	scheme   *runtime.Scheme
+	clock    clock.Clock
+	recorder record.EventRecorder
+}
+
+// +kubebuilder:rbac:groups=apps.kruise.io,resources=imagelistpulljobs,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=apps.kruise.io,resources=imagelistpulljobs/status,verbs=get;update;patch
+
+// Reconcile reads that state of the cluster for a ImageListPullJob object and makes changes based on the state read
+// and what is in the ImageListPullJob.Spec
+// Automatically generate RBAC rules to allow the Controller to read and write ImageListPullJob
+func (r *ReconcileImageListPullJob) Reconcile(_ context.Context, request reconcile.Request) (res reconcile.Result, err error) {
+	klog.V(5).Infof("Starting to process ImageListPullJob %v", request.NamespacedName)
+
+	// 1.Fetch the ImageListPullJob instance
+	job := &appsv1alpha1.ImageListPullJob{}
+	err = r.Get(context.TODO(), request.NamespacedName, job)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Object not found, return.  Created objects are automatically garbage collected.
+			// For additional cleanup logic use finalizers.
+			scaleExpectations.DeleteExpectations(request.String())
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, err
+	}
+
+	// If resourceVersion expectations have not satisfied yet, just skip this reconcile
+	resourceVersionExpectations.Observe(job)
```

Comment:
I think you can remove the resourceVersionExpectation for ImageListPullJob, and add resourceVersionExpection for ImagePullJob.

Reply:
Sure , and will add the retry function when update status . as you said above.